client/components/date-range/index.js

```diff
@@ -0,0 +1,526 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindAll, noop, isNil } from 'lodash';
+import { DateUtils } from 'react-day-picker';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { localize, moment } from 'i18n-calypso';
+import DatePicker from 'components/date-picker';
+import Popover from 'components/popover';
+import DateRangeInputs from './inputs';
+import DateRangeHeader from './header';
+import DateRangeTrigger from './trigger';
+
+export class DateRange extends Component {
+	static propTypes = {
+		selectedStartDate: PropTypes.oneOfType( [
+			PropTypes.instanceOf( Date ),
+			PropTypes.instanceOf( moment ),
+		] ),
+		selectedEndDate: PropTypes.oneOfType( [
+			PropTypes.instanceOf( Date ),
+			PropTypes.instanceOf( moment ),
+		] ),
+		onDateSelect: PropTypes.func,
+		onDateCommit: PropTypes.func,
+		firstSelectableDate: PropTypes.oneOfType( [
+			PropTypes.instanceOf( Date ),
+			PropTypes.instanceOf( moment ),
+		] ),
+		lastSelectableDate: PropTypes.oneOfType( [
+			PropTypes.instanceOf( Date ),
+			PropTypes.instanceOf( moment ),
+		] ),
+		triggerText: PropTypes.func,
+		renderTrigger: PropTypes.func,
+		renderHeader: PropTypes.func,
+		renderInputs: PropTypes.func,
+	};
+
+	static defaultProps = {
+		onDateSelect: noop,
+		onDateCommit: noop,
+		renderTrigger: props => <DateRangeTrigger { ...props } />,
+		renderHeader: props => <DateRangeHeader { ...props } />,
+		renderInputs: props => <DateRangeInputs { ...props } />,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		// Define the date range as Moment instances
+		const firstSelectableDate = ! isNil( this.props.firstSelectableDate )
+			? this.props.moment( this.props.firstSelectableDate )
+			: undefined;
+		const lastSelectableDate = ! isNil( this.props.lastSelectableDate )
```

Comment:
I might be misinterpreting the usage of `isNil` here, but would the usage of `has` work for us here?: 
```js
const lastSelectableDate = has( this.props, 'lastSelectableDate' ) && this.props.moment( this.props.firstSelectableDate );
```
If so we can avoid a double negative (`! isNil` is equal to `is not nil`) and better show intent.
Just a thought though, I could be missing something there.

Reply:
Seems to work. I used the original because I thought we'd need `undefined` explicitly to "turn off" the feature in the `DatePicker` rather than `false` which `has()` returns. But your solution works just fine and is much clearer as to the intent. I was never comfortable with a ternary condition resulting in `undefined`. It just looked wrong.

Fixed.